RockWeb/Webhooks/Mailgun.ashx

```diff
@@ -78,19 +78,26 @@ public class Mailgun : IHttpHandler
             
             if ( !string.IsNullOrWhiteSpace( request.Form["workflow_action_guid"]))
             {
-                Guid? actionGuid = request.Form["workflow_action_guid"].AsGuidOrNull();
+                Guid? actionGuid = request.Form["workflow_action_guid"].Split( ',' )[0].AsGuidOrNull();
                 string status = string.Empty;
                 switch ( eventType )
                 {
-                    case "complained":
-                    case "unsubscribed":
+                    case "complained": break;
+                    case "unsubscribed": break;
                     case "delivered": status = SendEmailWithEvents.SENT_STATUS; break;
                     case "clicked": status = SendEmailWithEvents.CLICKED_STATUS; break;
                     case "opened": status = SendEmailWithEvents.OPENED_STATUS; break;
-                    case "dropped": 
-                    case "bounced": 
-                        status = SendEmailWithEvents.FAILED_STATUS; break;
-                }                
+                    case "dropped": break;
+                    case "bounced": status = SendEmailWithEvents.FAILED_STATUS;
+                        int secs = request.Form["timestamp"].AsInteger();
+                        DateTime ts = new DateTime( 1970, 1, 1, 0, 0, 0, DateTimeKind.Utc ).AddSeconds( secs ).ToLocalTime();
```

Comment:
Rather than ToLocalTime() (which will report the time the event occurred as UTC regardless of the Rock installation's time zone) you will want to use RockDateTime.OrgTimeZoneInfo to identify the offset that should be used.